### PR TITLE
Upgrade go version to v1.17

### PR DESCRIPTION
**Pull request by AxiomSamarth**
**What this PR does / why we need it**:
Update golang version to v1.17

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```other user
Updated golang version to v1.17
```

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,5 +1,5 @@
 #############      builder                                  #############
-FROM eu.gcr.io/gardener-project/3rd/golang:1.15.5 AS builder
```
Comment (himanshu-kun) [inline Dockerfile:2]: The base image can also be updated to 
`eu.gcr.io/gardener-project/3rd/alpine:3.13.7` as done in [this](https://github.com/gardener/machine-controller-manager-provider-openstack/pull/44) PR

File: go.mod [hunk 2]
```diff
@@ -14,3 +14,67 @@ require (
 	k8s.io/klog v0.4.0
 	k8s.io/utils v0.0.0-20201110183641-67b214c5f920
 )
+
+require (
+	github.com/Microsoft/go-winio v0.4.16 // indirect
+	github.com/ProtonMail/go-crypto v0.0.0-20210428141323-04723f9f07d7 // indirect
+	github.com/acomagu/bufpipe v1.0.3 // indirect
+	github.com/beorn7/perks v1.0.1 // indirect
```
Comment (himanshu-kun) [inline go.mod:22]: we can combine the 2 `require` blocks...

**Review by himanshu-kun — CHANGES_REQUESTED** [review]
lgtm
some minor clean code changes and others requested

File: go.mod (hunk 2 above)
Comment (AxiomSamarth) [reply go.mod:22]: Running `make revendor` or `go mod vendor` will separate them again. That's how go v1.17 is designed to treat the indirect dependencies. Read more [here](https://golang.org/doc/go1.17#go-command)

File: Dockerfile (hunk 1 above)
Comment (himanshu-kun) [reply Dockerfile:2]: Please update the image to `3.15.0` , as `3.13.7` doesn't exist

**Review by himanshu-kun — APPROVED** [review]
/lgtm

**Pull request merged**